update.php

```diff
@@ -739,12 +788,27 @@ function pre_update_1364()
 
 function pre_update_1365()
 {
-	if (!DBA::e("DELETE FROM `notify-threads` WHERE NOT `notify-id` IN (SELECT `id` FROM `notify`)")) {
+	if (DBStructure::existsTable('notify-threads') &&
+		!DBA::e("DELETE FROM `notify-threads` WHERE NOT `notify-id` IN (SELECT `id` FROM `notify`)")) {
 		return Update::FAILED;
 	}
 
-	if (!DBA::e("DELETE FROM `thread` WHERE NOT `iid` IN (SELECT `id` FROM `item`)")) {
+	if (DBStructure::existsTable('thread') &&
+		!DBA::e("DELETE FROM `thread` WHERE NOT `iid` IN (SELECT `id` FROM `item`)")) {
+		return Update::FAILED;	}
+
+	return Update::SUCCESS;
+}
+
+function pre_update_1368()
+{
+	// Lock cleanup to avoid constraint violations during foreign key creation
+	if (DBStructure::existsTable('locks') &&
```

Comment:
@annando do you think, this works as expected? Or do we possibly brick something when deleting locks which are maybe in use?

Reply:
The system **should** automatically go in maintenance during update. So nothing should be executed, so it should be no problem.